### PR TITLE
[stable/velero] Configure Velero plugins with ConfigMaps

### PR DESCRIPTION
**Pull request by sebastien-prudhomme**
#### What this PR does / why we need it:

My use case is to change the restic restore image with a configmap as documented here: https://velero.io/docs/v1.0.0/restic/

I've taken a generic approach that can be used for other Velero plugins instead of a dedicated configmap for the restic restore image. Give me feedback if it's not a good idea.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/velero/README.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Parameter | Description | Default
 `metrics.serviceMonitor.enabled` | Set this to `true` to create ServiceMonitor for Prometheus operator | `false`
 `metrics.serviceMonitor.additionalLabels` | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus | `{}`
 `schedules` | A dict of schedules | `{}`
+`configmaps` | Configure Velero plugins with ConfigMaps | `[]`
```
Comment (hectorj2f) [inline stable/velero/README.md]: Use `configMaps` to match the property name

Comment (hectorj2f) [reply stable/velero/README.md]: Please, change the description to Velero ConfigMaps. It is commonly used for plugins but that isn't the unique purpose.

File: stable/velero/templates/configmaps.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{{- range .Values.configMaps }}
+---
+apiVersion: v1
+kind: ConfigMap
+{{ toYaml . }}
```
Comment (hectorj2f) [inline stable/velero/templates/configmaps.yaml]: I'd suggest to keep a generic and static structure for the `metadata`  and `data` properties such as:

```
metadata:
  name: {{ .name }}
  labels:  {{ .labels }}
data:
 {{ .data }}
```

**Review by hectorj2f — CHANGES_REQUESTED** [review]

File: stable/velero/README.md (hunk 1 above)
Comment (sebastien-prudhomme) [reply stable/velero/README.md]: In my first draft version it was called ```pluginsConfigMaps```. Perhaps using that name is more clear?

File: stable/velero/templates/configmaps.yaml (hunk 2 above)
Comment (sebastien-prudhomme) [reply stable/velero/templates/configmaps.yaml]: This was the way i've done it in my draft version. I was not sure about that so i changed it. The problem i see with that is that if user want to add annotations, or other things, the chart need to be updated.

Before applying your suggestion, i will check other charts to see if there exists a common pattern for that generic way of adding resources.

Thank for your feedback!

File: stable/velero/README.md (hunk 1 above)
Comment (hectorj2f) [reply stable/velero/README.md]: let's keep it generic `configMaps` as you initially did, analogously to other charts.

File: stable/velero/templates/configmaps.yaml (hunk 2 above)
Comment (hectorj2f) [reply stable/velero/templates/configmaps.yaml]: > The problem i see with that is that if user want to add annotations, or other things, the chart need to be updated.

Good point. We could also add an annotation's field too:
```
{{- with .Values.xxx.annotations }}
  annotations:
{{ toYaml . | indent 4 }}
{{- end }}
```

Comment (hectorj2f) [reply stable/velero/templates/configmaps.yaml]: Otherwise, just skip the annotations field and let others add it, if required.

Comment (sebastien-prudhomme) [reply stable/velero/templates/configmaps.yaml]: I've seen how it's done in ```templates/schedules.yaml```

What do you think of something like that (not tested yet) :

```
{{- range .Values.configMaps }}
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ include "velero.fullname" $ }}-{{ .name }}
  labels:
    app.kubernetes.io/name: {{ include "velero.name" $ }}
    app.kubernetes.io/instance: {{ $Release.Name }}
    app.kubernetes.io/managed-by: {{ $Release.Service }}
    helm.sh/chart: {{ include "velero.chart" $ }}
  {{- with .labels }}
    {{- toYaml .labels | nindent 4 }}
  {{- end }}
data:
  {{- toYaml .data | nindent 2 }}
---
{{- end }}
```

The good things are the standard labels and a name that is now build on the release name

File: stable/velero/README.md (hunk 1 above)
Comment (sebastien-prudhomme) [reply stable/velero/README.md]: Done

File: stable/velero/templates/configmaps.yaml (hunk 2 above)
Comment (sebastien-prudhomme) [reply stable/velero/templates/configmaps.yaml]: I've finally implemented, and tested, something like the solution i suggested.

**Comment by aleks-mariusz** [comment]
as helm deployment name is prefixed with the release name, velero plugin add fails because it tries to find the (release-less) deployment by name `velero`.. 

can you demonstrate how one would use this configmap to install say the portworx-velero plugin?

**Comment by sebastien-prudhomme** [comment]
@aleks-mariusz the configmap is not for installing plugins but for configuring them. In the documentation only the embedded restic plugin seems to use that kind on configmap.

It seems there is no mechanism yet in the chart to do the "velero plugin add" command that install plugins.

**Comment by sebastien-prudhomme** [comment]
@aleks-mariusz after reading the whole values.yaml, you certainly can install the plugin by configuring `initContainers`

**Pull request merged**